File: problems_051to075/problem_055/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch
from toolbox import is_palindrome


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 0

        iteration_limit = 49

        for starter in range(1, 10000):
            num = starter
            iteration_count = 0

            while iteration_count < iteration_limit:
                digits = [digit for digit in str(num)]
                digits.reverse()
                reversed_num = int(''.join(digits))
                num += reversed_num

                if is_palindrome(str(num)):
                    break

                iteration_count += 1

            if iteration_count == iteration_limit:
                answer += 1

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_051to075/problem_055/solution.js
const { isPalindrome } = require('../../lib/toolbox');

const solution = () => {
  let answer = null;

  const iterationLimit = 49;

  for (let starter = 1; starter < 10000; starter += 1) {
    let num = starter;
    let iterationCount = 0;

    while (iterationCount < iterationLimit) {
      const reversedNum = Number(String(num).split('').reverse().join(''));

      num += reversedNum;

      if (isPalindrome(String(num))) {
        break;
      }

      iterationCount += 1;
    }

    if (iterationCount === iterationLimit) {
      answer += 1;
    }
  }

  return answer;
};

if (process.env.NODE_ENV !== 'test') {
  // eslint-disable-next-line no-console
  console.log(solution());
}

module.exports = solution;
